core/src/main/java/org/elasticsearch/rest/action/cat/RestSnapshotAction.java

```diff
@@ -0,0 +1,115 @@
+
+
+import org.elasticsearch.action.admin.cluster.snapshots.get.GetSnapshotsRequest;
+import org.elasticsearch.action.admin.cluster.snapshots.get.GetSnapshotsResponse;
+import org.elasticsearch.client.Client;
+import org.elasticsearch.common.Table;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.rest.RestChannel;
+import org.elasticsearch.rest.RestController;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.rest.RestResponse;
+import org.elasticsearch.rest.action.support.RestResponseListener;
+import org.elasticsearch.rest.action.support.RestTable;
+import org.elasticsearch.snapshots.SnapshotInfo;
+import org.joda.time.format.DateTimeFormat;
+import org.joda.time.format.DateTimeFormatter;
+
+import java.util.concurrent.TimeUnit;
+
+import static org.elasticsearch.rest.RestRequest.Method.GET;
+
+/**
+ * Cat API class to display information about snapshots
+ */
+public class RestSnapshotAction extends AbstractCatAction {
+    @Inject
+    public RestSnapshotAction(Settings settings, RestController controller, Client client) {
+        super(settings, controller, client);
+        controller.registerHandler(GET, "/_cat/snapshots/{repository}", this);
+    }
+
+    @Override
+    protected void doRequest(final RestRequest request, RestChannel channel, Client client) {
+        GetSnapshotsRequest getSnapshotsRequest = new GetSnapshotsRequest();
+        getSnapshotsRequest.repository(request.param("repository"));
+        getSnapshotsRequest.snapshots(new String[] { GetSnapshotsRequest.ALL_SNAPSHOTS });
+        getSnapshotsRequest.masterNodeTimeout(request.paramAsTime("master_timeout", getSnapshotsRequest.masterNodeTimeout()));
+
+        client.admin().cluster().getSnapshots(getSnapshotsRequest, new RestResponseListener<GetSnapshotsResponse>(channel) {
+            @Override
+            public RestResponse buildResponse(GetSnapshotsResponse getSnapshotsResponse) throws Exception {
+                return RestTable.buildResponse(buildTable(request, getSnapshotsResponse), channel);
+            }
+        });
+    }
+
+    @Override
+    protected void documentation(StringBuilder sb) {
+        sb.append("/_cat/snapshots/{repository}\n");
+    }
+
+    @Override
+    protected Table getTableWithHeader(RestRequest request) {
+        return new Table()
+                .startHeaders()
+                .addCell("id", "alias:id,snapshotId;desc:unique snapshot id")
+                .addCell("status", "alias:s,status;text-align:right;desc:snapshot name")
+                .addCell("start_epoch", "alias:ste,startEpoch;desc:start time in seconds since 1970-01-01 00:00:00")
+                .addCell("start_time", "alias:sti,startTime;desc:start time in HH:MM:SS")
+                .addCell("end_epoch", "alias:ete,endEpoch;desc:end time in seconds since 1970-01-01 00:00:00")
+                .addCell("end_time", "alias:eti,endTime;desc:end time in HH:MM:SS")
```

Comment:
Can we also have a `duration` or a `elapsed` milliseconds? We can go ahead and do the math for people.